src/pages/UserPreferences/index.tsx

```diff
@@ -0,0 +1,313 @@
+          title: 'Unable to find the target registry.',
+        });
+        return;
+      }
+      registries.splice(rowIndex, 1);
+    }
+    this.setState({ isDeleteModalOpen: false, currentRegistryIndex: -1 });
+    try {
+      if (rowIndex === undefined) {
+        this.setState({ selectedItems: [] });
+      }
+      await this.props.updateDockerRegistries(registries);
+      this.showAlert({
+        key: `delete-registr${rowIndex !== undefined ? 'y' : 'ies'}-success`,
+        variant: AlertVariant.success,
+        title: `Registr${rowIndex !== undefined ? 'y' : 'ies'} successfully deleted.`,
+      });
+    } catch (e) {
+      this.showAlert({
+        key: `delete-registr${rowIndex !== undefined ? 'y' : 'ies'}-fail`,
+        variant: AlertVariant.danger,
+        title: `Unable to delete the registr${rowIndex !== undefined ? 'y' : 'ies'}. ${e}`,
+      });
+    }
+  }
+
+  private setEditModalStatus(isEditModalOpen: boolean): void {
+    if (this.state.isEditModalOpen === isEditModalOpen) {
+      return;
+    }
+    this.setState({ isEditModalOpen });
+  }
+
+  private setDeleteModalStatus(isDeleteModalOpen: boolean): void {
+    if (this.state.isDeleteModalOpen === isDeleteModalOpen) {
+      return;
+    }
+    this.setState({ isDeleteModalOpen });
+  }
+
+  private async onRegistriesChange(registries: RegistryRaw[], isEditMode: boolean): Promise<void> {
+    this.setState({ isEditModalOpen: false, currentRegistryIndex: -1 });
+    try {
+      await this.props.updateDockerRegistries(registries);
+      this.showAlert({
+        key: `${isEditMode ? 'edit' : 'add'}-registry-success`,
+        variant: AlertVariant.success,
+        title: `Registry successfully ${isEditMode ? 'saved' : 'added'}.`,
+      });
+    } catch (e) {
+      this.showAlert({
+        key: `${isEditMode ? 'edit' : 'add'}-registry-fail`,
+        variant: AlertVariant.danger,
+        title: `Unable to ${isEditMode ? 'saved' : 'added'} the registry. ${e}`,
+      });
+    }
+  }
+
+  render(): React.ReactNode {
+    const { isLoading } = this.props;
+    const { isEditModalOpen, isDeleteModalOpen, currentRegistryIndex, registries, selectedItems } = this.state;
+    const columns = ['Host', 'Username'];
+    const rows = this.state.registries.map(registry => ({
+      cells: this.buildRegistryRow(registry),
+      selected: registry.selected,
+    })) || [];
+    const actions = [
+      { title: 'Edit registry', onClick: (event, rowIndex) => this.showOnEditRegistryModal(rowIndex) },
+      { title: 'Delete registry', onClick: (event, rowIndex) => this.showOnDeleteRegistryModal(rowIndex) },
+    ];
+
+    return (
+      <React.Fragment>
+        <Head pageName="User Preferences" />
+        <PageSection variant={PageSectionVariants.light}>
+          <Title headingLevel={'h1'}>User Preferences</Title>
+          <TextContent>
+            <Text component={TextVariants.p}>
+              Control the configuration and policies for your installation.
```

Comment:
It seems to be outdated description but I don't have a good suggestion on how it should look like. On the suggested design of external accounts, it's just placeholder.

Can we just remove it until we don't have a good idea what should be here?

@sympatheticmoose Do you have an idea on that part?
![](https://user-images.githubusercontent.com/6310786/104624285-0f820c00-569c-11eb-97a8-fff1bbf54a1d.png)

Reply:
@sleshchenko Remove seems sensible for now whilst it's just the single tab. Curious why we've switched from Container Registry/Registries to Docker?

Reply:
![](https://media.giphy.com/media/RkthcGH9Jmt52QZQUt/giphy.gif)

We didn't! It's just a typo :-D

@olexii4 please fix it

Reply:
I have fixed it.

Reply:
@olexii4 just checking you also saw the comment that we should be referring to Container Registry / Registries rather than Docker?

Reply:
@sympatheticmoose  I have fixed it.

Reply:
Awesome. Thanks @olexii4